colossalai/nn/graph/utils.py

```diff
@@ -0,0 +1,50 @@
+import functools
+import torch
+from colossalai.tensor import ColoTensor
+from typing import Callable, List
+from colossalai.nn._ops._utils import convert_to_colo_tensor
+
+
+def register_colo_graph(input_pos: List[int], param_pos: List[int]) -> Callable:
```

Comment:
I think purely relying on `__torch_function__` to trace the computation graph has some limitations, e.g. memory consumption due to tensor materialization, does not capture control flow and some operations such as indexing like `out = inputs[0]`.

Reply:
We have implemented a `Proxy` to fulfill the requirements above.

Reply:
OK, I see. This method can be covered by proxy. 
I want to make some quick changes to ColoTensor. For example, I want to build a table that contains <Op, start, finish> for all activation tensors and give hints for their memory allocation for inference.

Reply:
Not exactly, an index in the topological graph.

Reply:
ok.

Reply:
I might need some coarse-grained information on the graph to try out my ideas. 
I also plan to be involved in the development of fx later.